src/common/Database.js

```diff
@@ -183,7 +183,13 @@
     };
 
     Grid.prototype.fieldByLabel = function (_, ignoreCase) {
-        return this.fields().filter(function (field, idx) { field.idx = idx; return ignoreCase ? field.label().toLowerCase() === _.toLowerCase() : field.label() === _; })[0];
+        var retVal = this.fields().filter(function (field, idx) { field.idx = idx; return ignoreCase ? field.label().toLowerCase() === _.toLowerCase() : field.label() === _; })[0];
+        if (!retVal) {
+            console.log("Not able to find field: " + _);
+            console.log(this.fields());
+            return {};
```

Comment:
@GordonSmith we need to fix the DDL/ or something then so that a field isnt requested just yet before its fetched? basically we dont want to be throwing an error ... the error is fixed after the data and index is fetched ... @jbrundage can you email us the DDL this fails on?